### PR TITLE
#341 Implement editable node labels within canvas

### PR DESCRIPTION
**Pull request by tomlyn**
Fixes: #341 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/src/command-actions/setNodeLabelAction.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017-2020 Elyra Authors
```
Comment (caritaou) [inline canvas_modules/common-canvas/src/command-actions/setNodeLabelAction.js:2]: Its 2021 now :)

File: canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js [hunk 2]
```diff
@@ -2985,6 +2993,57 @@ export default class SVGCanvasRenderer {
 		this.logger.logEndTimer("displayNodes " + this.getFlags());
 	}
 
+	displayEditIcon(spanObj, node) {
```
Comment (caritaou) [inline canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js:2996]: Is there a better way to handle this situation?
image

Maybe some auto wrapping of the text to the next line for preview?

File: canvas_modules/common-canvas/src/command-actions/setNodeLabelAction.js (hunk 1 above)
Comment (tomlyn) [reply canvas_modules/common-canvas/src/command-actions/setNodeLabelAction.js]: I fixed all the copyrights for files I've changed and created.

File: canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js (hunk 2 above)
Comment (tomlyn) [reply canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js:2996]: This problem only occurs when the 'show full label on hover' option is switched on .  I fixed the code so the edit icon appears at the end of the hover over text.

**Review by caritaou — APPROVED** [review]

**Pull request merged**